Civi/Api4/Generic/AbstractEntity.php

```diff
@@ -86,7 +86,9 @@ protected static function getEntityName() {
    * @return string
    */
   protected static function getEntityTitle($plural = FALSE) {
-    return static::getEntityName();
+    $name = static::getEntityName();
+    $dao = \CRM_Core_DAO_AllCoreTables::getFullName($name);
+    return $dao ? $dao::getEntityTitle($plural) : ($plural ? \CRM_Utils_String::pluralize($name) : $name);
```

Comment:
So, if I'm reading correctly, this is an overall reduction in the number of functions, and it works because `CRM_Core_DAO_AllCoreTables::getFullName` is fairly gentle about errors (ie returns `NULL` for unrecognized/non-DAO entities).

Reply:
Yes I moved it here because the `RelationshipCache` entity does not extend `DAOEntity`. Ditto for the `getInfo` function.